controllers/promiserelease_controller.go

```diff
@@ -225,3 +228,29 @@ func (r *PromiseReleaseReconciler) reconcileOnInstalledPromise(o opts, promiseRe
 
 	return ctrl.Result{}, nil
 }
+
+func (r *PromiseReleaseReconciler) validateVersion(o opts, promiseRelease *v1alpha1.PromiseRelease, promise *v1alpha1.Promise) (updated bool, err error) {
+	promiseVersion, found := promise.GetLabels()[promiseVersionLabel]
+	if !found {
+		return false, fmt.Errorf("version label (%s) not found on promise; refusing to install", promiseVersionLabel)
+	}
+
+	if promiseRelease.Spec.Version == "" {
+		promiseRelease.Spec.Version = promiseVersion
+		err := o.client.Update(o.ctx, promiseRelease)
+		if err != nil {
+			return false, fmt.Errorf("failed to set promise release version: %w", err)
+		}
+		return true, nil
+	}
+
+	if promiseVersion != promiseRelease.Spec.Version {
+		return false, fmt.Errorf(
+			"version label on promise (%s) does not match version on promise release (%s); refusing to install",
+			promiseVersion,
+			promiseRelease.Spec.Version,
+		)
+	}
+
+	return false, nil
+}
```

Comment:
do we wan to add support for some sort of `latest`, `any`/blank or `*` option? i imagine that would be useful for dev

Reply:
i think for dev you may just drop the promise-release altogether.

i think adding support for those would be good though, just not sure we need to do it now (there will be an entire epic for floating versions i suppose)